### ISSUE TITLE
Fine Tuned MMS Model produces Gibberish

### ISSUE THREAD

**Issue by RohitMidha23** [facebookresearch/fairseq#5309]
HI, 

I'm fine-tuning the MMS large model using Gujarati data. I have approximately `400` hours of training data and have been using that. I trained the adapter layers for `5 epochs`, with a starting learning rate of `5e-6`. Train loss is going down as I'm training but when I try to predict, I get gibberish. Here is a sample output: 

```
<s>અ<s>ર<s>્<s>થ<s>ા<s>ત<s> <s>એ<s>ન<s>ું <s>ગ<s>ઞ<s>ા<s>ન<s>',
 '<s>અ<s>ન<s>ે<s> <s>ત<s>મ<s>ે<s> <s>જ<s>ો<s>સ<s>ો<s> <s>શ<s>ા<s>ં<s>ત<s>િ<s> <s>આ<s>ન<s>ં<s>દ<s> <s>શ<s>ુ<s>દ<s>્<s>ધ<s>િ<s> <s>ત<s>મ<s>ા<s>ર<s>ી<s> <s>અં<s>દ<s>ર<s> પ<s>્ર<s>ગ<s>ટ<s> <s>થ<s>ી<s> <s>શ<s>ે<s> <s>પ<s>જ<s>ી<s>તો<s> <s>આ<s>ખ<s>ુ<s>ં<s> <s>ફ<s>a<s>m<s>i<s>l<s>ી<s> <s>છ<s>ે<s> <s>ક<s>ર<s>ુ<s>ન<s>ા<s> <s>i<s>ત<s>્<s>ય<s>ા<s>દ<s>ી<s>અ<s>ન<s>ે<s> <s>મ<s>ુ<s>ખ<s>્<s>ય<s>ત<s> <s>શ<s>ુ<s>દ<s>્<s>ધ<s>િ<s> <s>શ<s>ા<s>ં<s>ત<s>િ<s>ન<s>ે<s> <s>આ<s>ન<s>ં<s>દ<s> <s>પ<s>્<s>ર<s>ગ<s>ટ<s>ત<s>ા<s> જ<s>શ<s>ે<s> <s>અ<s>ન<s>ે<s> <s>ત<s>ે<s>થ<s>ી<s> <s>પ<s>ા<s>પ<s>ન<s>ો<s> <s>આ<s>ધ<s>ા<s>ર<s> <s>છ<s>ૂ<s>ટ<s>ટ<s>ો<s> જ<s>ે<s>શ<s>ે<s>',
 '<s>l<s>i<s>t<s>e<s>c<s>o<s>મ<s>e<s>s<s> <s>d<s>a<s>c<s>t<s>r<s>i<s>n<s>s<s> <s>d<s>i<s>s<s>a<s>p<s>e<s>a<s>r<s>s<s> <s>સ<s>્<s>વ<s>ર<s>ૂ<s>પ<s> <s>બ<s>ો<s>ધ<s> <s>જ<s>ે<s>ટ<s>લ<s>ો<s> <s>આ<s>વ<s>સ<s>ે<s> જ<s>ી<s>વ<s>ન<s>મ<s>ા<s>ં<s> <s>એ<s>ટ<s>લ<s>ુ<s>ં<s> <s>પ<s>ા<s>પ<s>ન<s>ો<s> <s>આ<s>ધ<s>ા<s>ર<s> <s>જ<s>ુ<s>ટ<s>ત<s>ો<s> <s>જ<s>ે<s>શ<s>ે<s> <s>વ<s>િ<s>ભ<s>ા<s>વ<s>ો<s>ન<s>ી<s> <s>શ<s>ક<s>્<s>ત<s>િ<s> <s>મ<s>ં<s>દ<s>મ<s>ં<s>દ<s>મ<s>ં<s>દ<s>બ<s>ં<s>દ<s> <s>થ<s>ત<s>ા<s> <s>પ<s>છ<s>ી<s> <s>ક<s>ો<s>ઈ<s> <s>વ<s>િ<s>ક<s>ા<s>ર<s>ન<s>ે<s> <s>ક<s>ો<s>ઈ<s> <s>વ<s>િ<s>ક<s>ષ<s>ે<s>પ<s> <s>ન<s>ે<s> <s>ડ<s>ે<s> <s>વ<s>િ<s>ક<s>l<s>્<s>પ<s> <s>મ<s>ા<s>ત<s>્<s>ર<s>ન<s>ે<s> <s>સ<s>ો<s> <s>દ<s>i<s>s<s> <s>h<s>a<s>s<s> <s>t<s>o<s> <s>બ<s>e<s> <s>આ<s> <s>n<s>્ય<s>ુ<s>y<s>r<s>s<s> <s>t<s>a<s>r<s>ગ<s>e<s>t<s> ક<s>ે<s> <s>શ<s>ુ<s>દ<s>્<s>ધ<s> <s>સ<s>્<s>વ<s>ર<s>ૂ<s>પ<s> <s>ત<s>ર<s>િ<s>ખ<s>ા<s>લ<s>ી<s> <s>સ<s>્<s>વ<s>ર<s>ૂ<s>પ<s>ન<s>ુ<s>ં<s>
```

This is on an audio clip of around 1 min. 

Can someone tell me what I'm doing wrong / suggest possible solutions. 

Any help on the same would be greatly appreciated.

**Comment by vineelpratap** [comment]
Hi, please check this tutorial on how to finetune MMS models - https://huggingface.co/blog/mms_adapters

**Comment by RohitMidha23** [comment]
@vineelpratap that is the exact blog post I'm using as reference. I've also looked at and tried 
 [run_speech_recognition_ctc_adapter.py](https://github.com/huggingface/transformers/blob/main/examples/pytorch/speech-recognition/run_speech_recognition_ctc_adapter.py)

**Comment by vineelpratap** [comment]
Hi, please double check everything to make sure data preparation, token dictionary and training is being done correctly. I would expect it to work for Gujarati language without any issues. 

You could possibly start with making sure 
1. You are able to run the tutorial on the same language (Turkish) and the same dataset. 
2. Then modify the data to Gujarati and try to run the finetuning again.


Also, you mentioned that the training loss is going down. Are you also able to see validation CER/WER going down as well ?

**Comment by RohitMidha23** [comment]
@vineelpratap i realised there is an issue with a portion of my data that is causing the model to produce such output. Thanks for your help!
